cohortextractor/patients.py

```diff
@@ -108,6 +241,40 @@ def most_recent_bmi(
     The date of the measurement can be obtained using `date_of("<bmi-column-name>")`.
     If the BMI is computed from weight and height then we use the date of the
     weight measurement for this.
+
+    Args:
+        on_or_before (str, None): String of Data in YYYY-MM-DD format or None,
```

Comment:
(hasn't read the doc docs fully yet...) is there a way to abstract this out for reuse?